Packs/Logzio/Integrations/Logzio/Logzio.py

```diff
@@ -0,0 +1,295 @@
+    response = requests.request("POST", url, headers=headers, data=json.dumps(payload))
+    if response.status_code != 200:
+        return_error('Error in API call [%d] - %r: %r' % (response.status_code, response.reason, response.content))
+    return response
+
+
+def test_module(client):
+    try:
+        client.fetch_triggered_rules()
+        return 'ok'
+    except Exception as e:
+        return 'Test failed: {}'.format(e)
+
+
+def search_logs_command(client, args):
+    query = args.get('query')
+    size = args.get('size', 1000)
+    from_time = args.get('from_time')
+    to_time = args.get('to_time')
+    resp = client.search_logs(query, size, from_time, to_time)
+    content = [res["_source"] for res in resp]
+    return {
+        'ContentsFormat': formats['json'],
+        'Type': entryTypes['note'],
+        'Contents': content,
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': tableToMarkdown("Logs", content),
+        'EntryContext': {
+            'Logzio.Logs.Count': len(content),
+            'Logzio.Logs.Results': content
+        }
+    }
+
+
+def search_logs_by_fields_command(client, args):
+    size = args.get('size', 1000)
+    from_time = args.get('from_time', "")
+    to_time = args.get('to_time', "")
+    params = {}
+    for i in range(1, 3):
+        params[args.get('key%s' % i)] = args.get('value%s' % i)
+    remove_nulls_from_dictionary(params)
+    query = " AND ".join(["{}:\"{}\"".format(key, params[key]) for key in params])
+    resp = client.search_logs(query, size, from_time, to_time)
+    content = [res["_source"] for res in resp]
+    return {
+        'ContentsFormat': formats['json'],
+        'Type': entryTypes['note'],
+        'Contents': content,
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': tableToMarkdown("Logs", content),
+        'EntryContext': {
+            'Logzio.Logs.Count': len(content),
+            'Logzio.Logs.Results': content
+        }
+    }
+
+
+def get_rule_logs_by_id_command(client, args):
+    id = args.get("id")
+    size = args.get("size", 100)
+    page_size = args.get("page_size", 1000)
+    resp = client.get_rule_logs(id, size, page_size)
+    return {
+        'ContentsFormat': formats['json'],
+        'Type': entryTypes['note'],
+        'Contents': resp,
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': tableToMarkdown("Logs", resp),
+        'EntryContext': {
+            'Logzio.Logs.Results': resp
+        }
+    }
+
+
+def fetch_incidents(client, last_run, search, severities, first_fetch_time):
+    incidents = []
+    next_run = last_run
+    start_query_time = last_run.get("last_fetch")
+    if not start_query_time:
+        start_query_time, _ = parse_date_range(first_fetch_time, date_format=DATE_FORMAT, utc=False, to_timestamp=True)
+        start_query_time = start_query_time / 1000
+        next_run["last_fetch"] = max(start_query_time, time.time() - ONE_HOUR)
```

Comment:
This is not affecting the first/retro-active fetch. Sometimes our the ingestion in our system might take a few minutes, so you can think of it as a retry mechanism for an hour, in case that happens.

Reply:
i might be missing something, but it seems to be this condition will be true only on the first/retro-active fetch

Reply:
That's right, and it's exactly what it handles

Reply:
but that way we ignore `start_query_time` if its more than one hour

Reply:
We only set an initial value to the **next run** field, if this first run will return results, the fields will be overwritten with the timestamp of the most recent incident. If not, it will be the latest of the one hour/retro-active fetch

Reply:
ok got it now
why do we need that? does it matter to you if it queries, for example the last 24 hours, or last one hour ?

Reply:
Minimizing resource consumption. Our system is Elastic Search based, we strive to search in fewer indexes as possible.